### DISCUSSION TITLE
Connection errors when connecting to a Supabase database

### DISCUSSION THREAD

**Discussion by inian** [supabase/supabase#20951] [Troubleshooting]
If you receive any of the following error messages, it might be because the network you are connecting from does not support IPv6. 

- `(dial tcp [2001:db8:3333:4444:5555:6666:7777:8888]:5432: connect: no route to host)`
- `connect to db.example.supabase.co (2001:db8:3333:4444:5555:6666:7777:8888) port 5432 (tcp) failed: Network is unreachable`
- `could not translate host name "db.example.supabase.co" to address: nodename nor servname provided, or not known`
- `ENETUNREACH 2001:db8:3333:4444:5555:6666:7777:8888`
- `Error: P1001: Can't reach database server at db.example.supabase.co:5432`
- `(2001:db8:3333:4444:5555:6666:7777:8888), port 5432 failed: could not create socket: Address family not supported by protocol`

Note that these errors may manifest in cases other than your client network not supporting IPv6, but if you run into these errors after your project was migrated, it is likely that it is due to your network not supporting IPv6.

To remediate this issue, either 
- configure your network for IPv6.
- connect via Supavisor which resolves to a IPv4 address. You can get to your Supavisor connecting from the database settings page [here](https://supabase.com/dashboard/project/_/settings/database).
- Purchase [the IPv4 addon](https://supabase.com/docs/guides/platform/ipv4-address).
- Switch to using our client libraries or PostgREST.

Visit [this discussion](https://github.com/orgs/supabase/discussions/17817) for more information about the IPv6 transition.

**Comment by WillyRamirez** [comment]
I am receiving the first error locally when I run `supabase link`  as described [here](https://supabase.com/docs/guides/functions/deploy#link-your-local-project). How can I proceed if my network does not support Ipv6?

**Reply by mtjosue** [reply]
Hello, I am getting the Error: P1001: Can't reach database server.
I am trying to update the tables in db with prisma, using prisma db push. 
Since I am using the connection URI on the .env of the project with the password. This is the "connect via Supavisor" way. 
I also tried the ?pgbouncer=true&connection_limit=1 appending from the prisma connection guide.
Just trying to update tables in db. Anything helps. thanks

**Comment by mtjosue** [comment]
Hello, I am getting the Error: P1001: Can't reach database server.
I am trying to update the tables in db with prisma, using prisma db push. 
Since I am using the connection URI on the .env of the project with the password. This is the "connect via Supavisor" way. 
I also tried the ?pgbouncer=true&connection_limit=1 appending from the prisma connection guide.
Just trying to update tables in db. Anything helps. thanks
